pandas/tseries/offsets.py

```diff
@@ -2748,14 +2751,20 @@ def nanos(self):
 
     def apply(self, other):
         # Timestamp can handle tz and nano sec, thus no need to use apply_wraps
-        if isinstance(other, (datetime, np.datetime64, date)):
+        if isinstance(other, Timestamp):
+            result = other.__add__(self)
```

Comment:
Do you really need to call `__add__` here rather than using `+`?

Generally calling double-underscore methods directly is an anti-pattern, so if so this definitely deserves a comment to explain.